src/Functions/array/arrayElement.cpp

```diff
@@ -472,11 +472,24 @@ ColumnPtr FunctionArrayElement::executeNumberConst(
     auto col_res = ColumnVector<DataType>::create();
 
     if (index.getType() == Field::Types::UInt64)
+    {
         ArrayElementNumImpl<DataType>::template vectorConst<false>(
             col_nested->getData(), col_array->getOffsets(), safeGet<UInt64>(index) - 1, col_res->getData(), builder);
+    }
     else if (index.getType() == Field::Types::Int64)
+    {
+        /// Cast to UInt64 before negation allows to avoid undefined behaviour for negation of the most negative number.
+        /// NOTE: this would be undefined behaviour in C++ sense, but nevertheless, compiler cannot see it on user provided data,
```

Comment:
AFAIK, it's never an undefined behaviour according to the standard. Conversion from negative signed to unsigned integer is defined as addition of `2^{N-1}` and negation of unsigned integer `value` is defined as `2^N - value`. Am I missing something?

Reply:
It is the comment explaining why do we have this code (this code prevents UB that will be otherwise possible).